src/FluffySpoon.AspNet.LetsEncrypt/LetsEncryptRenewalService.cs

```diff
@@ -273,31 +272,37 @@ private async Task ValidateOrderAsync(string[] domains, IOrderContext order)
 
 				var challengeExceptions = nonNullChallengeValidationResponses
 					.Where(x => x.Status == ChallengeStatus.Invalid)
-					.Select(x => new Exception($"{x.Error.Type}: {x.Error.Detail} (challenge type {x.Type})"))
+					.Select(x => new Exception($"{x.Error?.Type ?? "errortype null"}: {x.Error?.Detail ?? "null errordetails"} (challenge type {x.Type ?? "null"})"))
 					.ToArray();
 
-				if (challengeExceptions.Length > 0)
-					throw new OrderInvalidException(
+				if (challengeExceptions.Length > 0) 
+                {
+					var exception = new OrderInvalidException(
 						"One or more LetsEncrypt orders were invalid. Make sure that LetsEncrypt can contact the domain you are trying to request an SSL certificate for, in order to verify it.",
 						new AggregateException(challengeExceptions));
+
+                    _logger.LogTrace("Throwing exception: ", exception);
```

Comment:
Can we get rid of this? Shouldn't the exception in itself be useful?

Reply:
Done